SELearning.API/Controllers/ContentController.cs

```diff
@@ -20,33 +20,54 @@ public ContentController(ILogger<ContentController> logger, IContentRepository r
         _repository = repository;
     }
 
+    /// <summary>
+    /// <c>GetContent</c> returns the content with the given ID.
+    /// </summary>
+    /// <param name="ID">The ID of the content.</param>
+    /// <returns>A content with the given ID if it exists, otherwise response type 404: Not Found.</returns>
     [Authorize]
     [HttpGet("{ID}")]
-    [ProducesResponseType(typeof(ContentDTO), 200)] // OK
-    [ProducesResponseType(404)] // Not Found
+    [ProducesResponseType(typeof(ContentDTO), 200)]
+    [ProducesResponseType(404)]
     public async Task<ActionResult<ContentDTO>> GetContent(string ID)
         => (await _repository.GetContent(ID)).ToActionResult();
 
+    /// <summary>
+    /// <c>CreateContent</c> creates a content.
+    /// </summary>
+    /// <param name="content">The record of the content.</param>
+    /// <returns>A response type 201: Created.</returns>
     [Authorize]
     [HttpPost]
-    [ProducesResponseType(201)] // Created
+    [ProducesResponseType(201)]
     public async Task<IActionResult> CreateContent(ContentDTO content)
     {
         var (result, created) = await _repository.AddContent(content);
         return CreatedAtRoute(nameof(GetContent), new { created.ID }, created);
     }
 
+    /// <summary>
+    /// <c>UpdateContent</c> updates the content with the given ID.
+    /// </summary>
+    /// <param name="ID">The ID of the content.</param>
+    /// <param name="content">The record of the updated content.</param>
+    /// <returns></returns>
     [Authorize]
     [HttpPut("{ID}")]
-    [ProducesResponseType(204)] // No Content
-    [ProducesResponseType(404)] // Not Found
+    [ProducesResponseType(204)]
+    [ProducesResponseType(404)]
     public async Task<IActionResult> UpdateContent(string ID, ContentDTO content)
         => (await _repository.UpdateContent(ID, content)).ToActionResult();
 
+    /// <summary>
+    /// <c>GetContent</c> deletes the content with the given ID, and its associated comments.
```

Comment:
Does `DeleteContent` also delete the associated comments of the content that is deleted?

Reply:
It should. But i think we should set it up on a database level using on delete cascade. Which judging from [this](https://docs.microsoft.com/en-us/ef/core/saving/cascade-delete) seems to be the default behavior. So the answer is yes i guess.